### ISSUE TITLE
azurerm_cosmosdb_mongo_collection existence of default_ttl_seconds key causes to reset indexes

### ISSUE THREAD

**Issue by hokugawa** [terraform-providers/terraform-provider-azurerm#9301]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.29

azurerm 2.6.0

### Affected Resource(s)

* `azurerm_cosmosdb_mongo_collection`

### Terraform Configuration Files

```hcl
resource "azurerm_cosmosdb_mongo_collection" "foo" {
  name                = "foo"
  resource_group_name = azurerm_cosmosdb_account.db_account.resource_group_name
  account_name        = azurerm_cosmosdb_account.db_account.name
  database_name       = azurerm_cosmosdb_mongo_database.database.name

  default_ttl_seconds = "2592000" # 30 days
  shard_key           = "uuid"
  throughput          = var.mongo_collection_foo_throughput

  timeouts {}
}
```

### Expected Behaviour

terraform should not modify existing indexes.

### Actual Behaviour

no changes for mongo collection indexes.

Please note that I'm still using 2.6.0. I know that the newer version of azurerm supports the `index` block, but I feel that it is not reliable enough yet. Hence, I stick to the 2.6.0

### Steps to Reproduce

1. Create cosmosdb mongodb collection by terraform
1. Create mongodb collection indexes by using mongo shell (e.g. `db.foo.createIndex({uuid:1})`, etc)
1. Run `terraform apply` again
1. Check indexes by using mongo shell (e.g. `db.foo.getIndexes()`)

If `azurerm_cosmosdb_mongo_collection` has `default_ttl_seconds` key, it will reset indexes.

At least, terraform document should note this behavior. Because losing indexes would cause a serious problem.

### References

* #0000

**Comment by yupwei68** [comment]
Hi @hokugawa , thanks for opening this issue. 
Unfortunately, we have upgraded the cosmos service version from `2015-04-08` to `2020-04-01` in v2.20.0. 
Currently, I could not reproduce this problem in the latest release of Azurerm v2.37.0. I'd like to suggest a version upgrade later than "v2.20.0" for you.

**Comment by hokugawa** [comment]
Hi @yupwei68 Thanks for the reply.
With newer versions such as v2.37.0, it requires adding "_id" index in the `index` block. 

> Error: Error waiting on create/update future for Cosmos Mongo Collection "..." (Account: "...", Database: "..."): Code="BadRequest" Message="Index definition does not contains '_id' index specification.\r\nActivityId: ..., Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0"

However, currently, I can't add all necessary indexes in the index block due to #9045 

So, I created indexes by using mongo shell command, but when I run the terraform, it will erase indexes that not on the `index` block.

I couldn't upgrade the provider version because of this issue.

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 4.x version of the provider please do let us know by opening a new issue, thanks!
